### ISSUE TITLE
Where in WCAG do you record issues in touch swipe focussing and activation of controls?

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#1907]
We are currently discussing where to record issues arising when testing apps on mobile devices with screen reader on, using gestural input (swipe focussing and double tap activation). 
https://github.com/BIK-BITV/BIK-App-Test/issues/23
So this issue is not so much wrong sequence or inert text elements (this we would place in 1.3.2 and 1.3.1) but interactive controls that cannot be focused and/or activated with gestures with screen reader on. So far we have recorded this under 2.1.1 Keyboard and 2.4.3 Focus Order and, where role, name status are missing, obviously in 4.1.2. Name, Role, Value. It doesn't seem right but are there better places for issues that you obviously need to capture when testing on touch devices?
I'd be interested to know how other people deal with these issues and where they place them.

PS: The problem currently surfaces in testing native apps where we have no source code access, but maybe the question extends also to web apps.

**Comment by patrickhlauke** [comment]
indeed, this is one that always bubbles up and leaves folks confused most of the time. my very general take is usually "focus order". do you also test just with keyboard but without AT? if those controls fail in that case as well, obviously 2.1.1. it does get trickier when/if it does work with keyboard, but for some reason doesn't with AT running (for instance if the control/container has been hidden from AT altogether). then you start to cross into 4.1.2 and possibly even 1.3.1 (stretching it perhaps, but in a  "information is visually conveyed but not programmatically exposed" way).

**Comment by jake-abma** [comment]
agree with @patrickhlauke ands wondering if you have live examples @detlevhfischer where keyboard works, but SR not?

**Comment by patrickhlauke** [comment]
it gets even more interesting when it's not necessarily controls, but just static content. then you can't ding it normatively under 2.1.1 nor 2.4.3. leaves 1.3.1 in my view, with the above rationale. same with static/non-interactive content that is hidden from AT using reading keys on non-touch/keyboard ATs

**Comment by mraccess77** [comment]
Generally I'd flag interactive content that is keyboard accessible but not accessible with AT under SC 4.1.2.      For content that is not in the reading order with AT but on-screen I would flag 1.3.2 or 1.3.1 depending on what I thought the cause of the issue was.  For example, if I could explore by touch and get info out of it probably 1.3.2 but if explore by touch doesn't work then probably 1.3.1 as it may be marked as decorative/hidden from AT.

**Comment by detlevhfischer** [comment]
> if you have live examples @detlevhfischer where keyboard works, but SR not?

Its a running audit, so I am reluctant to share it. Mind you, keyboard accessibility on iPad might involve moving the focus via up/down and left/right arrows, not in sequential tab or arrowing order, so technically, yes, controls can be reached but 2.4.3 is likely a fail (since it is in effect 2D navigation where you need sigh to get the focus to the right spot).

**Comment by detlevhfischer** [comment]
> if you have live examples @detlevhfischer where keyboard works, but SR not?

The added complication is that on iOS, some (arguably the more important) functions of an non-focusable menu icon may be available on neighbouring elements with rotor on "container" via vertical swipe to an action and then double tab activation, but not **all** of the functions offered in said popup menu when called up... the funny world of mobile a11y testing.

**Comment by detlevhfischer** [comment]
@mraccess77 
> Generally I'd flag interactive content that is keyboard accessible but not accessible with AT under SC 4.1.2. 

4.1.2 for me doea not seem quite right when you want to capture the issue that something cannot be activated. It is very much like 2.1.1 Keyboard, but concerned with another mode of input. Maybe a future requirement covering also SR activation would call for "Mode-Independent activation" (I guess there is no word "activatability", right?)

**Comment by JAWS-test** [comment]
@detlevhfischer 

> 4.1.2 for me does not seem quite right when you want to capture the issue that something cannot be activated.

4.1.2 has:

> states, properties, and values that can be set by the user can be programmatically set

That is why operating problems that occur with AT are a problem with 4.1.2

**Comment by patrickhlauke** [comment]
yes but you don't "set" any property/state/value when you're, say, trying to activate a link or button. (and going further with non-interactive content, that'd then be completely irrelevant if you can't reach it with AT navigation). at a stretch i can see the rationale for checkboxes/radios/toggles, but it's not a panacea here

**Comment by fstrr** [comment]
This issue looks like it can be closed. If it needs to be re-opened, please do that and convert it to a Discussion.
